### PR TITLE
Version 4.4.0 bug fix and features

### PR DESCRIPTION
**Pull request by redJ4y**
## Features:
  - Added `RaygunHandler.from_sender()` factory to construct a `RaygunHandler` instance using an existing `RaygunSender`. This allows for additional configuration of the sender.
  - Added a `config` parameter the to Flask and WSGI middleware provider constructors. This also allows for additional configuration of the sender.
  - The `RaygunHandler` now adds tags corresponding to the logging level, which now defaults to `logging.ERROR`.
  - Errors/exceptions sent via the `RaygunHandler` now have their message overriden by the logged message.

## Bug fixes:
  - The `RaygunHandler` now attempts to capture `exc_info` from the `record`. This can be obtained if `logger.exception()` is used or if `exc_info=True` is set in the logger call.
    - If `exc_info` cannot be obtained by the `RaygunHandler`, it no longer attempts to construct a `RaygunErrorMessage` with `None` values. Instead, it generates a fallback error message using information gathered from the `record`. This is essentially an error with a single stack frame representing the call to the logger.
 
## Quality of life updates:
  - Updated `CONTRIBUTING.MD`.
  - Got unit tests running again (`django` upgrade).
  - Updated `python3/samples/sample.py` and `python3/samples/sampleWithLogging.py`.
  - Cleaned up `python3/raygun4py/cli.py`.

### CODE REVIEW

File: python3/raygun4py/raygunmsgs.py [hunk 1]
```diff
@@ -93,19 +98,19 @@ def set_tags(self, tags):
...
+            self.raygunMessage.details['tags'] += tags
         return self
 
     def set_request_details(self, request):
         if not request:
             return self
 
-        rg_request_details = http_utilities.build_wsgi_compliant_request(request)
+        rg_request_details = http_utilities.build_wsgi_compliant_request(
+            request)
```
Comment (Olwiba) [inline python3/raygun4py/raygunmsgs.py:109]: this line wrap ok?

**Review by Olwiba — COMMENTED** [review]
I've completed a high-level review on this and this is looking really nice, great work @redJ4y 💪 
Appreciate the in-code documentation & quality of code improvements.

I see Panos will give it a review also so I'll let him apply the approval

File: python3/raygun4py/raygunmsgs.py (hunk 1 above)
Comment (redJ4y) [reply python3/raygun4py/raygunmsgs.py]: That does look pretty silly, no idea what autopep8 was thinking!

File: python3/raygun4py/middleware/wsgi.py [hunk 2]
```diff
@@ -1,48 +1,53 @@
...
 log = logging.getLogger(__name__)
 
 
 class Provider(object):
 
-    def __init__(self, app, apiKey):
+    def __init__(self, app, apiKey, config=None):
         self.app = app
-        self.sender = raygunprovider.RaygunSender(apiKey)
+        self.config = config if config is not None else {}
```
Comment (Hamish-taylor) [inline python3/raygun4py/middleware/wsgi.py:12]: Does `config` need to be stored in this object? could it just be created locally and then passed into the `.RaygunSender(...)`? Same note for the flask middleware

**Review by Hamish-taylor — COMMENTED** [review]
Excellent work! The changes seem good to me, I have just left one small question. I also did not run the code so make sure you have tested your changes well

File: python3/raygun4py/middleware/wsgi.py (hunk 2 above)
Comment (Hamish-taylor) [reply python3/raygun4py/middleware/wsgi.py:12]: Resolved as this is needed for unit test

**Review by Hamish-taylor — APPROVED** [review]
Nice work! 🚀

**Pull request merged**